web/app/js/components/Sidebar.jsx

```diff
@@ -115,17 +120,36 @@ class Sidebar extends React.Component {
     }
   }
 
+  handleNamespaceSelector(value) {
+    this.setState({namespaceFilter: value});
+  }
+
+  filterResourcesByNamespace(resources, namespace) {
+    if (namespace === "all") {
+      return resources;
+    }
+
+    let result = _.mapValues(resources, o => {
+      return _(o)
+        .filter(r => r.resource.namespace === namespace)
+        .value();
+    });
+    return result;
+  }
+
   render() {
     let normalizedPath = this.props.location.pathname.replace(this.props.pathPrefix, "");
     let PrefixedLink = this.api.PrefixedLink;
-    let numHiddenNamespaces = _.size(this.state.namespaces) - this.state.maxNsItemsToShow;
+    let namespaces = [{nsValue: "all", nsName: "All Namespaces"}].concat(_.map(this.state.namespaces, ns => {
+      return {nsValue: ns, nsName: ns};
+    }));
+    let sidebarComponents = this.filterResourcesByNamespace(this.state.resourceGroupings, this.state.namespaceFilter);
```

Comment:
authorities are kind of different from the other resources and it makes less sense to have this type of dashboard for them; also they're not linked in the web server routes to going to `http://localhost:8084/namespaces/linkerd/authoritys/10.1.1.97:9995` won't go anywhere. We should filter out the resource types that we don't have resource detail pages for (or add routing for them in `web/srv/server.go`, currently there's only three resource types there)